src/sentry/notifications/notifications/rules.py

```diff
@@ -90,12 +92,16 @@ def get_context(self) -> MutableMapping[str, Any]:
         alert_status_page_enabled = features.has(
             "organizations:alert-rule-status-page", self.project.organization
         )
+        alert_type = str(AlertRuleTriggerAction.Type.EMAIL.name).lower()
+        alert_timestamp = str(round(time.time() * 1000))
+        rules = get_rules(self.rules, self.organization, self.project)
+        alert_rule_id = str(rules[0].id) if rules and rules[0].id else None
         context = {
             "project_label": self.project.get_full_name(),
             "group": self.group,
             "event": self.event,
-            "link": get_link(self.group, environment),
-            "rules": get_rules(self.rules, self.organization, self.project),
+            "link": get_link(self.group, environment, alert_type, alert_timestamp, alert_rule_id),
```

Comment:
`get_link()` is only being called on this line so I think it'd be worth pushing 

```py
alert_type = str(AlertRuleTriggerAction.Type.EMAIL.name).lower()
alert_timestamp = str(round(time.time() * 1000))
alert_rule_id = str(rules[0].id) if rules and rules[0].id else None

```
into the function and calling it `get_group_settings_link()`.

Reply:
Incorporated your suggestions in new commit.